assets/js/modules/tagmanager/components/setup/SetupForm.js

```diff
@@ -16,64 +16,124 @@
 
 /**
  * Internal dependencies
  */
 import Data from 'googlesitekit-data';
-import Button from '../../../../components/button';
 import { STORE_NAME, FORM_SETUP, EDIT_SCOPE } from '../../datastore/constants';
+import { STORE_NAME as MODULES_ANALYTICS } from '../../../analytics/datastore/constants';
 import { STORE_NAME as CORE_FORMS } from '../../../../googlesitekit/datastore/forms/constants';
 import { STORE_NAME as CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { STORE_NAME as CORE_MODULES } from '../../../../googlesitekit/modules/datastore/constants';
 import { isPermissionScopeError } from '../../../../googlesitekit/datastore/user/utils/is-permission-scope-error';
 import {
 	AccountSelect,
 	AMPContainerSelect,
-	FormInstructions,
 	WebContainerSelect,
 } from '../common';
-import StoreErrorNotice from '../../../../components/StoreErrorNotice';
+import Button from '../../../../components/button';
+import Link from '../../../../components/link';
+import SetupErrorNotice from './SetupErrorNotice';
+import SetupFormInstructions from './SetupFormInstructions';
 const { useSelect, useDispatch } = Data;
 
-export default function SetupForm( { finishSetup } ) {
+export default function SetupForm( { finishSetup, setIsNavigating } ) {
 	const canSubmitChanges = useSelect( ( select ) => select( STORE_NAME ).canSubmitChanges() );
+	const singleAnalyticsPropertyID = useSelect( ( select ) => select( STORE_NAME ).getSingleAnalyticsPropertyID() );
+	const analyticsModuleActive = useSelect( ( select ) => select( CORE_MODULES ).isModuleActive( 'analytics' ) );
 	const hasEditScope = useSelect( ( select ) => select( CORE_USER ).hasScope( EDIT_SCOPE ) );
-	const autoSubmit = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_SETUP, 'autoSubmit' ) );
+	const analyticsModuleReauthURL = useSelect( ( select ) => select( MODULES_ANALYTICS ).getAdminReauthURL() );
+	// Only select the initial autosubmit + submitMode once from form state which will already be set if a snapshot was restored.
+	const initialAutoSubmit = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_SETUP, 'autoSubmit' ), [] );
+	const initialSubmitMode = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_SETUP, 'submitMode' ), [] );
 
 	const { setValues } = useDispatch( CORE_FORMS );
+	const { activateModule } = useDispatch( CORE_MODULES );
 	const { submitChanges } = useDispatch( STORE_NAME );
-	const submitForm = useCallback( async ( event ) => {
-		event.preventDefault();
-		const { error } = await submitChanges();
-		if ( isPermissionScopeError( error ) ) {
-			setValues( FORM_SETUP, { autoSubmit: true } );
-		}
-		if ( ! error ) {
+	const dispatchAnalytics = useDispatch( MODULES_ANALYTICS );
+	const submitForm = useCallback( async ( { submitMode } = {} ) => {
+		const throwOnError = async ( func ) => {
+			const { error } = await func() || {};
+			if ( error ) {
+				throw error;
+			}
+		};
```

Comment:
This would obviously be useful elsewhere as well but it's defined here for now as a best practice for hook callbacks as there isn't a clear place for this to go just yet. Alternatively, creating something like `createStrictDispatch` could be useful to create decorated bound actions. The return value is our convention here so we can't magically make it work with everything.